engine/src/test/java/com/redhat/cloud/notifications/exports/ExportEventListenerTest.java

```diff
@@ -0,0 +1,640 @@
+import com.redhat.cloud.notifications.MockServerLifecycleManager;
+import com.redhat.cloud.notifications.TestLifecycleManager;
+import com.redhat.cloud.notifications.db.repositories.EventRepository;
+import com.redhat.cloud.notifications.exports.transformers.TransformersHelpers;
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.mockito.InjectMock;
+import io.smallrye.reactive.messaging.providers.connectors.InMemoryConnector;
+import io.smallrye.reactive.messaging.providers.connectors.InMemorySource;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.mockserver.integration.ClientAndServer;
+import org.mockserver.model.HttpRequest;
+import org.mockserver.model.MediaType;
+
+import javax.enterprise.inject.Any;
+import javax.inject.Inject;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.time.LocalDate;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static com.redhat.cloud.notifications.TestConstants.DEFAULT_ORG_ID;
+import static com.redhat.cloud.notifications.exports.ExportEventListener.EXPORT_CHANNEL;
+import static com.redhat.cloud.notifications.exports.ExportEventListener.FILTER_DATE_FROM;
+import static com.redhat.cloud.notifications.exports.ExportEventListener.FILTER_DATE_TO;
+import static org.awaitility.Awaitility.await;
+import static org.mockserver.model.HttpRequest.request;
+import static org.mockserver.model.HttpResponse.response;
+
+@QuarkusTest
+@QuarkusTestResource(TestLifecycleManager.class)
+public class ExportEventListenerTest {
+
+    private static final LocalDate TODAY = LocalDate.now(ZoneOffset.UTC);
+
+    @ConfigProperty(name = "export-service.psk")
+    String exportServicePsk;
+
+    @InjectMock
+    EventRepository eventRepository;
+
+    @Inject
+    ExportEventListener exportEventListener;
+
+    @Any
+    @Inject
+    InMemoryConnector inMemoryConnector;
+
+    @Inject
+    MicrometerAssertionHelper micrometerAssertionHelper;
+
+    /**
+     * Sets up the routes for the Mock Server.
+     */
+    @BeforeEach
+    void setUpMockServerRoutes() {
+        final ClientAndServer mockServer = MockServerLifecycleManager.getClient();
+
+        mockServer
+            .when(request().withPath(".*/error"))
+            .respond(response().withStatusCode(200));
+
+        mockServer
+            .when(request().withPath(".*/upload"))
+            .respond(response().withStatusCode(200));
```

Comment:
Testing with MockServer works, but it requires more work than tests based on `@InjectMock @RestClient` and Mockito:
https://github.com/RedHatInsights/notifications-backend/blob/35a969b12e921e38260427f35d8f749a6d5258d3/backend/src/test/java/com/redhat/cloud/notifications/routers/EndpointResourceTest.java#L2078-L2085

Reply:
Do you plan on keeping the MockServer approach here? I'm not requesting a change, I only want to make sure this class is in its final state before reviewing it in details.

Reply:
Yes, I'm going to start working on this one now

Reply:
Done! I've left the Mock Server just to test that the "Content-Type" headers are correct for each method.